### PR TITLE
Generate classes from preset slugs in the same way (server & client)

### LINKED ISSUE #32347: Different class names for preset slugs
Themes provide some presets to the editor, such as font sizes and colors, either via the `add_theme_support` https://github.com/WordPress/gutenberg/issues/26940 or the `theme.json` https://github.com/WordPress/gutenberg/issues/31629

The slug of the preset values is used to generate a class name in two different situations:

1. when in the editor, to generate the class name to be attached to the block markup upon an user action, as in:

```html
<p class="has-slug-preset">Content</p>
```

2. to generate the classes to be enqueued as part of the global styles stylesheet:

```css
.has-slug-preset { ... }
```

## What's the issue

When the slug contains a number, the class attached to the block markup and the class generated for the stylesheet are different. See issues reported at https://github.com/WordPress/gutenberg/issues/31629 and https://github.com/WordPress/gutenberg/issues/26940 There's also issues with slugs that contain camelCased words.

Slug | Class generated in the editor | Class generated in the front
--- | --- | --- 
`slug1` | `.has-slug-1-preset` | `.has-slug1-preset`
`2slug` | `.has-2-slug-preset` | `.has-2slug-preset`
`slug3number` | `.has-slug-3-number-preset` | `.has-slug3number-preset`
`4th` | `.has-4th-preset` | `.has-4th-preset`
`5sl` | `.has-5-sl-preset` | `.has-5sl-preset`
`slugNumber` | `.has-slug-number-preset` | `.has-slugnumber-preset`

Notice how the server doesn't process the slug but the client splits by number, except if the slug is an English ordinal such as `4th`. The client also converts camelCased slugs into kebab-cased slugs, while the server doesn't.

## Why this happens

In the client, both for font-sizes and colors we use the lodash's [kebabCase](https://lodash.com/docs/4.17.15#kebabCase) function to process the incoming slug such as:

```js
has-${ kebabCase( slug ) }-preset
```

The lodash's kebabCase function has the particularity that it considers numbers as something to separate as well, [unless they're English ordinals](https://github.com/lodash/lodash/issues/2843#issuecomment-262255270).

In the server, we don't process the slug but output it as it is:

```php
'.has-' . $value['slug'] . '-' . $class['class_suffix']
```

## How to fix it

1. Do not do any processing to the slug in the client, but use the incoming value. Existing classes would still work: the old slug is recognized as an extra class and because the theme should have enqueued it, it'll still work.

2. Replicate the lodash's kebabCase logic (including English ordinals) in the server.

3. Use our own regexp / function to do the same in both places, to make sure changes to lodash's kebabCase don't affect us.

### PR DESCRIPTION
**Pull request by oandregal**
Fixes https://github.com/WordPress/gutenberg/issues/32347

### CODE REVIEW

File: packages/block-editor/src/components/colors/test/utils.js [hunk 1]
```diff
@@ -108,11 +108,5 @@ describe( 'color utils', () => {
 				getColorClassName( 'background', undefined )
 			).toBeUndefined();
 		} );
-
-		it( 'should return a class name with the color slug in kebab case', () => {
-			expect( getColorClassName( 'background', 'Light Purple' ) ).toBe(
```
Comment (ntsekouras) [inline packages/block-editor/src/components/colors/test/utils.js:113]: So what happens with cases like this with `spaces`? Wouldn't they break this by creating more css classes?

Comment (oandregal) [reply packages/block-editor/src/components/colors/test/utils.js:113]: https://github.com/WordPress/gutenberg/issues/32347#issuecomment-851579181 :) I think we should make sure the classes don't have spaces. There are a few options here, I'm inclined to converting them to `-`.

Comment (oandregal) [reply packages/block-editor/src/components/colors/test/utils.js:113]: Fixed at https://github.com/WordPress/gutenberg/pull/32352/commits/def68c83fecdf6d3190b66098c5a4b488908cc87 including folding multiple consecutive spaces into a single `-`.

**Review by ntsekouras — APPROVED** [review]
LGTM 👍 - thanks!

**Pull request merged**